index.js

```diff
@@ -0,0 +1,104 @@
+// JavaScript code for calculator functionality
+const resultInput = document.getElementById('result')
+const buttons = document.querySelectorAll('.calculator button')
+
+// Add click event listeners to the calculator buttons
+buttons.forEach(button => {
+  button.addEventListener('click', () => {
+    const buttonText = button.textContent
+    handleButtonClick(buttonText)
+  })
+})
+
+// Function to handle button clicks
+function handleButtonClick (buttonText) {
```

Comment:
Check your calculate functions because currently they all return 'Error' for any calculations I do. Nice UI and I like your thinking. You could also check functions like `eval`